### ISSUE TITLE
Breadcrumb listitem added twice

### ISSUE THREAD

**Issue by AroundtheGlobe** [spatie/schema-org#84]
```php
foreach($row['breadcrumb'] as $breadcrumb) {
       $arrTemp[] = Schema::listItem()
              ->addProperties(['position' => $i])
              ->item(
                     Schema::webPage()
                            ->name($breadcrumb['title'])
                            ->url($breadcrumb['url'])
                            ->addProperties(['@id' => $breadcrumb['url']])
              );
       $i++;
}
$this->objSchema->breadcrumblist()->itemListElement($arrTemp);
```
When I try to add itemListElement to the breadcrumbs with the code above it seems to be listed twice. Am I using the code wrong or is there some sort of bug?

**Comment by Gummibeer** [comment]
What's `$this->objSchema`? And could you provide how you render the schema and what's your result?

**Comment by AroundtheGlobe** [comment]
```
        $this->objSchema      = new Graph();
        $this->objSchema->webpage()->name($row['title']);
        $this->objSchema->webpage()->inLanguage('nl');
        $this->objSchema->webpage()->description($row['description']);

        $this->objSchema->webpage()->datePublished(date("Y-m-d\TH:i:s+00:00", strtotime($row['date'])));

        $this->objSchema->breadcrumblist()->addProperties(array('@id' => Core::getCurrentUrl(false).'#breadcrumb'));
        $i = 1;
        $arrTemp = null;
        foreach($row['breadcrumb'] as $breadcrumb)
        {
            $arrTemp[] = Schema::listItem()->addProperties(array('position' => $i))->item(Schema::webPage()->name($breadcrumb['title'])->url($breadcrumb['url'])->addProperties(array('@id' => $breadcrumb['url'])));
            $i++;
        }
        $this->objSchema->breadcrumblist()->itemListElement($arrTemp);
        $this->objSchema->webpage()->breadcrumb($this->objSchema->breadcrumblist());
```
$row['breadcrumb'] contains 5 elements. 

The output (modified for readability)

```
"breadcrumb":
  {
  "@type":"BreadcrumbList",
  "@id":"#breadcrumb",
  "itemListElement":[{
      "@type":"ListItem",
      "position":1,
      "item":{
          "@type":"WebPage",
          "name":"Home",
          "url":"","@id":""
          }
     },
     {
     "@type":"ListItem",
     "position":2,
     "item": {"@type": "WebPage", "name": "Reizen","url": "", "@id": ""
     },
    "@type":"ListItem",
    "position":1,
    "item":{
    "@type":"WebPage",
    "name":"Home",
    "url":"","@id":""
    }
    },
    {
    "@type":"ListItem",
    "position":2,
    "item": {"@type": "WebPage", "name": "Reizen","url": "", "@id": ""
    }]
}
```

**Comment by Gummibeer** [comment]
Ok, I've rebuilt your example and everything is fine - but you can get confused pretty fast if you don't check exactly in which object/key you are.

```php
$row = [
    'title' => 'webpage title',
    'description' => 'webpage description',
    'date' => '2019-06-04 21:28:00',
    'breadcrumb' => [
        [
            'title' => 'Home',
            'url' => '',
        ],
        [
            'title' => 'Reizen',
            'url' => '',
        ],
    ],
];

$graph = new \Spatie\SchemaOrg\Graph();
$graph->webpage()
    ->name($row['title'])
    ->inLanguage('nl')
    ->description($row['description'])
    ->datePublished(new DateTime($row['date']));

$graph->breadcrumblist()->setProperty('@id', '#breadcrumb');

$i = 1;
$arrTemp = [];
foreach ($row['breadcrumb'] as $breadcrumb) {
    $arrTemp[] = \Spatie\SchemaOrg\Schema::listItem()
        ->setProperty('position', $i)
        ->item(
            \Spatie\SchemaOrg\Schema::webPage()
                ->name($breadcrumb['title'])
                ->url($breadcrumb['url'])
                ->setProperty('@id', $breadcrumb['url'])
        );
    $i++;
}

$graph->breadcrumblist()->itemListElement($arrTemp);
$graph->webpage()->breadcrumb($graph->breadcrumblist());
```

```json
{
  "@context":"https:\/\/schema.org",
  "@graph":[
    {
      "@type":"WebPage",
      "name":"webpage title",
      "inLanguage":"nl",
      "description":"webpage description",
      "datePublished":"2019-06-04T21:28:00+00:00",
      "breadcrumb":{
        "@type":"BreadcrumbList",
        "@id":"#breadcrumb",
        "itemListElement":[
          {
            "@type":"ListItem",
            "position":1,
            "item":{
              "@type":"WebPage",
              "name":"Home",
              "url":"",
              "@id":""
            }
          },
          {
            "@type":"ListItem",
            "position":2,
            "item":{
              "@type":"WebPage",
              "name":"Reizen",
              "url":"",
              "@id":""
            }
          }
        ]
      }
    },
    {
      "@type":"BreadcrumbList",
      "@id":"#breadcrumb",
      "itemListElement":[
        {
          "@type":"ListItem",
          "position":1,
          "item":{
            "@type":"WebPage",
            "name":"Home",
            "url":"",
            "@id":""
          }
        },
        {
          "@type":"ListItem",
          "position":2,
          "item":{
            "@type":"WebPage",
            "name":"Reizen",
            "url":"",
            "@id":""
          }
        }
      ]
    }
  ]
}
```

![screencapture-localhost-8080-test-php-2019-06-04-21_47_23](https://user-images.githubusercontent.com/6187884/58909047-65e24700-8712-11e9-8d36-500497656a17.png)

Your first breadcrumb is listed in `WebPage.BreadcrumbList` this is the result of `$graph->webpage()->breadcrumb($graph->breadcrumblist())`.
Your second breadcrumb is listed in `BreadcrumbList` which is the result of `$graph->breadcrumblist()->itemListElement($arrTemp)`.

If you want to hide the breadcrumb at top level you can use `$graph->hide(\Spatie\SchemaOrg\BreadcrumbList::class)`. Or you just keep booth or you don't add them to the graph.

**Comment by AroundtheGlobe** [comment]
Thank you a lot for your super quick support. I've played a very long time with the code again and after changing this: 
`$graph->webpage()->breadcrumb($graph->breadcrumblist());`
to
`$graph->webpage()->breadcrumb(array('@id' =>'#breadcrumb')); `

I got the desired result in the Google Schema testing tool. It turned out that the output of the library was correct, but the Google Schema Testing tool was incorrectly showing the data, possibly due to a mistake I made in the code.
